websubhub-compiler-plugin-test/src/test/java/io/ballerina/stdlib/websubhub/CompilerPluginTest.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.websubhub;
+
+import io.ballerina.projects.DiagnosticResult;
+import io.ballerina.projects.Package;
+import io.ballerina.projects.PackageCompilation;
+import io.ballerina.projects.ProjectEnvironmentBuilder;
+import io.ballerina.projects.directory.BuildProject;
+import io.ballerina.projects.environment.Environment;
+import io.ballerina.projects.environment.EnvironmentBuilder;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.PrintStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.MessageFormat;
+
+/**
+ * This class includes tests for Ballerina WebSub compiler plugin.
+ */
+public class CompilerPluginTest {
+    private static final Path RESOURCE_DIRECTORY = Paths
+            .get("src", "test", "resources", "ballerina_sources").toAbsolutePath();
+    private static final PrintStream OUT = System.out;
+    private static final Path DISTRIBUTION_PATH = Paths
+            .get("build", "target", "ballerina-distribution").toAbsolutePath();
+
+    @Test
+    public void testValidServiceDeclaration() {
+        Package currentPackage = loadPackage("sample_1");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.diagnostics().size(), 0);
+    }
+
+    @Test
+    public void testCompilerPluginForRemoteMethodValidation() {
+        Package currentPackage = loadPackage("sample_2");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.diagnostics().size(), 1);
+        Diagnostic diagnostic = (Diagnostic) diagnosticResult.diagnostics().toArray()[0];
+        DiagnosticInfo diagnosticInfo = diagnostic.diagnosticInfo();
+        Assert.assertNotNull(diagnosticInfo, "DiagnosticInfo is null for erroneous service definition");
+        Assert.assertEquals(diagnosticInfo.code(), "WEBSUBHUB_102");
+        String expectedMessage = "websubhub:Service should only implement remote methods";
```

Comment:
You can use the already declared constants for this (you would have to put the compiler-plugin module as a test dependency). It will be easier if you ever do any changes to the error messages or codes. **But this is also fine**. 
[Sample](https://github.com/ballerina-platform/module-ballerinax-nats/blob/5d3f69c1720cc0c41b97643cd3c39123fe484b5a/nats-compiler-plugin-tests/src/test/java/io/ballerina/stdlib/nats/plugin/NatsCompilerPluginTest.java#L100)

Reply:
Yeah, good to have constants here.